### DISCUSSION TITLE
Topology database when LSU packets arrive from the link that is down?

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#232] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
In pwospf.txt,
link state update and behavior on incoming LSU packet does not seem to discard LSU on links that are down (HELLO timeout).
How should such LSU packets be processed? Do we process as if the link is not down?
or when timeout happens, do we remove it from topology database?

**Comment by Pusnow** [comment, marked as answer]
Once a link failure detected (link state changes), router has to update the topology database (removing it) and floods LSU messages for the other routers to be get notified.

However, in PA4, we do not simulate link failures or state changes, so you don't need to handle them. You only need to focus on building topology database and Dijkstra's algorithm.
